e2e/_suites/fleet/fleet.go

```diff
@@ -139,28 +143,100 @@ func (fts *FleetTestSuite) afterScenario() {
 
 // beforeScenario creates the state needed by a scenario
 func (fts *FleetTestSuite) beforeScenario() {
+	maxTimeout := time.Duration(utils.TimeoutFactor) * time.Minute
+	exp := utils.GetExponentialBackOff(maxTimeout)
+
 	fts.StandAlone = false
 	fts.ElasticAgentStopped = false
 
 	fts.Version = common.BeatVersion
 
-	policy, err := fts.kibanaClient.GetDefaultPolicy(fts.currentContext, false)
-	if err != nil {
+	waitForPolicy := func() error {
+		policy, err := fts.kibanaClient.CreatePolicy(fts.currentContext)
+		if err != nil {
+			return errors.Wrap(err, "A new policy could not be obtained, retrying.")
+		}
+
 		log.WithFields(log.Fields{
-			"err": err,
-		}).Warn("The default policy could not be obtained")
+			"id":          policy.ID,
+			"name":        policy.Name,
+			"description": policy.Description,
+		}).Info("Policy created")
 
+		fts.Policy = policy
+
+		// Grab the system integration as we'll need to assign it a new name so it wont collide during
+		// multiple policy creations at once
+		integration, err := fts.kibanaClient.GetIntegrationByPackageName(context.Background(), "system")
+		if err != nil {
+			return err
+		}
+
+		packageDataStream := kibana.PackageDataStream{
+			Name:        fmt.Sprintf("%s-%s", integration.Name, uuid.New().String()),
+			Description: integration.Title,
+			Namespace:   "default",
+			PolicyID:    fts.Policy.ID,
+			Enabled:     true,
+			Package:     integration,
+			Inputs:      []kibana.Input{},
+		}
+
+		systemMetricsFile := filepath.Join(testResourcesDir, "/default_system_metrics.json")
+		jsonData := readJSONFile(systemMetricsFile)
+		for _, item := range jsonData.Children() {
+			if item.Path("type").Data().(string) == "system/metrics" {
+				packageDataStream.Inputs = append(packageDataStream.Inputs, kibana.Input{
+					Type:    item.Path("type").Data().(string),
+					Enabled: item.Path("enabled").Data().(bool),
+					Streams: item.S("streams").Data().([]interface{}),
+					Vars: map[string]kibana.Var{
+						"system.hostfs": {
+							Value: "",
+							Type:  "text",
+						},
+					},
+				})
+			} else {
+				packageDataStream.Inputs = append(packageDataStream.Inputs, kibana.Input{
+					Type:    item.Path("type").Data().(string),
+					Enabled: item.Path("enabled").Data().(bool),
+					Streams: item.S("streams").Data().([]interface{}),
+				})
+			}
+		}
```

Comment:
This is candidate to extract to a method, so that it's easier to be read

Reply:
Agreed, in a follow-up I wanted to support creating policy with optional integration added

Reply:
https://github.com/elastic/e2e-testing/issues/1853